pkg/cluster/cluster.go

```diff
@@ -346,7 +348,7 @@ func (c *Cluster) Create() error {
 	//
 	// Do not consider connection pooler as a strict requirement, and if
 	// something fails, report warning
-	if c.needConnectionPooler() {
+	if c.needMasterConnectionPooler() || c.needReplicaConnectionPooler() {
```

Comment:
I have suspicions that without distinction `needMasterConnectionPooler`/`needReplicaConnectionPooler` the code would be more clean. What do you thing about having only one function here for connection pool in general, and one helper function that returns the list of for what roles we need a pooler to be deployed? Similar to what is already implemented in some other place below in this pr.

Reply:
Seems like this is still a thing. I wanted to suggest removing `need{Master/Replica}ConnectionPooler` completely as they used mostly together anyway.

Reply:
Well, no it doesn't exist in the code anymore, it is also marked outdated. Maybe you accidentally had a look at old code.